cmd/posix.go

```diff
@@ -1243,41 +1217,57 @@ func (s *posix) CreateFile(volume, path string, fileSize int64, r io.Reader) (er
 		return remainingWritten, nil
 	}
 
-	var written int
+	bufp := s.pool.Get().(*[]byte)
+	defer s.pool.Put(bufp)
+
+	buf := *bufp
+
+	var written int64
 	for {
-		var n int
-		n, err = io.ReadFull(r, buf)
-		switch err {
-		case nil:
-			n, err = w.Write(buf)
-			if err != nil {
-				return err
+		nr, er := r.Read(buf)
+		if nr == len(buf) {
+			nw, ew := w.Write(buf)
+			if nw > 0 {
```

Comment:
we should check if `nw != len(buf)` instead, per golang docs for `os.Flle`'s `Write` method.